### PR TITLE
Implement manual snapshotting for RDS instances

### PR DESCRIPTION
**Pull request by jaredhm**
The backup functionality is more or less copied directly from our LTE project [here](https://github.com/massgov/mds-learn-to-earn/tree/develop/terraform/src/backup_lambda), albeit typescript-ified
The cleanup functionality is new, and has been tested locally

### CODE REVIEW

File: rdsinstance/src/backup_lambda/cli.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { handler } from "./index";
+
+const run = async () => {
+  process.env.RDS_INSTANCE_IDENTIFIER = process.env.RDS_INSTANCE_IDENTIFIER ?? 'itd-pr-pdg-reportingdata';
```
Comment (cs-ma) [inline rdsinstance/src/backup_lambda/cli.ts]: This seems odd - probably just left over from the original? If not, definitely don't think we should hardcode something like that in here

File: rdsinstance/src/cleanup_lambda/cli.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { handler } from "./index";
+
+const run = async () => {
+  process.env.RDS_INSTANCE_IDENTIFIER = process.env.RDS_INSTANCE_IDENTIFIER ?? 'itd-pr-pdg-reportingdata';
```
Comment (cs-ma) [inline rdsinstance/src/cleanup_lambda/cli.ts]: Same thing here?

File: rdsinstance/src/cleanup_lambda/index.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    if (dryRun) {
+      console.log(`DeleteDBSnapshot Message: ${JSON.stringify(params)}`)
+    } else {
+      result = await rds.deleteDBSnapshot(params).promise();
+    }
+    yield result;
+  }
+}
+
+const cleanupOldSnapShots = async (instances: Array<string>, dryRun = true): Promise<void> => {
```
Comment (cs-ma) [inline rdsinstance/src/cleanup_lambda/index.ts]: Is this empty function still needed?

**Review by cs-ma — CHANGES_REQUESTED** [review]
A few things I'm not sure about, otherwise looks good!

File: rdsinstance/src/backup_lambda/cli.ts (hunk 1 above)
Comment (jaredhm) [reply rdsinstance/src/backup_lambda/cli.ts]: Nah this is actually intentional. this is just a file for testing the handler locally in dry run mode

File: rdsinstance/src/cleanup_lambda/index.ts (hunk 3 above)
Comment (jaredhm) [reply rdsinstance/src/cleanup_lambda/index.ts]: nah lol my bad 🤦

File: rdsinstance/src/backup_lambda/cli.ts (hunk 1 above)
Comment (jaredhm) [reply rdsinstance/src/backup_lambda/cli.ts]: I can remove the hardcoded identifier though and make it a CLI arg

Comment (cs-ma) [reply rdsinstance/src/backup_lambda/cli.ts]: Nah that's fine!

**Review by cs-ma — APPROVED** [review]

**Pull request merged**